packages/react/src/components/List/List.jsx

```diff
@@ -155,6 +163,9 @@ const List = forwardRef((props, ref) => {
   const ListHeader = overrides?.header?.component || DefaultListHeader;
   const ListContent =
     overrides?.content?.component || isVirtualList ? VirtualListContent : DefaultListContent;
+  // getAllowedDropIds will be called by all list items when a drag is initiated and the
+  // paramater (id of the dragged item) will be the same until a new drag starts.
+  const memoizedGetAllowedDropIds = getAllowedDropIds ? memoize(getAllowedDropIds) : null;
```

Comment:
I wonder how well this scales.  If I have a list of 10k items, will it call this for 10k items?

Reply:
also, what is a virtual list and how is different than a normal list

Reply:
Since the `getAllowedDropIds` function is memoized and will receive the same param it will only be evaluated once per drag no matter how many items you have. 

For the old non virtual list with 10k items that doesn't use pagination the function `memoizedGetAllowedDropIds` will indeed be called 10k times but for such a list there are a lot of things that will happen 10k times and the calling of a memoized function would likely not be the straw that broke the camel's back.

The potential bottleneck lies in the number of items returned by the `getAllowedDropIds` since it is an array and not a map there will be a search for each rendered item's id. For large lists we now recommend using `isVirtualList` which means that only the items visible on the screen (I assume no more than around 30) will be calling and searching the result of  `memoizedGetAllowedDropIds` at the "same" time. Returning an array of 1000 ids does work but for more than that we probably need to support returning a map.

But even with `isVirtualList` and without the `getAllowedDropIds` our list component will be slow when using drag and drop in a list of 10k items. I also don't know what our UX designers would say about drag and drop in such a huge list even if it was fast though.

Reply:
The virtual list only renders the items currently visible, see https://github.com/bvaughn/react-window